### PR TITLE
BAU - create generic Navigator

### PR DESCRIPTION
**Pull request by SteveSugden**

### CODE REVIEW

File: app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/navigation/Navigator.scala [hunk 1]
```diff
@@ -16,72 +16,25 @@
...
-
-  protected val enterImporterEori: CreateAnswers => Boolean = (answers: CreateAnswers) =>
-    isRepresentative(answers) && answers.doesImporterHaveEori
-
-}
-
-protected trait Ordering {
-
-  protected val nextPageFor: (Seq[P], Page, CreateAnswers) => Option[Page] = (pages, currentPage, userAnswers) =>
+  protected val nextPageFor: (Seq[P], Page, T) => Option[Page] = (pages, currentPage, userAnswers) =>
```
Comment (crosslandwa) [inline app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/navigation/Navigator.scala:37]: `nextPageFor` can be private

```suggestion
  private val nextPageFor: (Seq[P], Page, T) => Option[Page] = (pages, currentPage, userAnswers) =>
```

File: app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/navigation/Navigator.scala [hunk 2]
```diff
@@ -99,4 +52,5 @@ protected trait Ordering {
     case s              => s.tail
   }
 
+  protected val always: Answers => Boolean = (_: Answers) => true
```
Comment (crosslandwa) [inline app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/navigation/Navigator.scala:55]: OBSERVATION: Moving `always` out from the `Conditions` trait makes the intended usage of this method by extending classes ambiguous IMO

I'd prefer one of
- moving `always` into `CreateAnswerConditions` (and duplicating it in `AmendAnswerConditions` when that exists)
- creating some sort of `SharedConditions` trait that is available for use by both `CreateNavigator` and `AmendNavigator`

**Review by crosslandwa — APPROVED** [review]
LGTM

FWIW i'd prefer to do this without extension (for fear of the `Navigator` trait attracting more complexity in the future to cater for specific extending classes), but I appreciate that (for now) having `CreateNavigator` and `AmendNavigator` _only_ have to define a page order and the access conditions is pretty neat/easy to understand

File: app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/navigation/Navigator.scala (hunk 2 above)
Comment (SteveSugden) [reply app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/navigation/Navigator.scala:55]: Is that an either/or suggestion?

Comment (crosslandwa) [reply app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/navigation/Navigator.scala]: Yes, its an either/or (though the first is the simpler one to implement while we only have one class extending the `Navigator` trait)

**Review by crosslandwa — DISMISSED** [review]

**Review by crosslandwa — APPROVED** [review]

**Pull request merged**